pkg/api/handler/default.go

```diff
@@ -63,15 +64,34 @@ func NewDefault(
 	peerRpcConfig *config.PeerRpcConfig,
 	apiConfig *config.ApiConfig,
 	opts ...Option,
-) (api.ServerInterface, error) {
+) (*ArcDefaultHandler, error) {
+
+	var wocClient *woc_client.WocClient
+	if apiConfig != nil {
+		wocClient = woc_client.New(woc_client.WithAuth(apiConfig.WocApiKey))
+	} else {
+		wocClient = woc_client.New()
+	}
+
+	finder := txFinder{
+		th:         transactionHandler,
+		pc:         peerRpcConfig,
+		l:          logger,
+		w:          wocClient,
+		useMainnet: true, // TODO: refactor in scope of ARCO-147
+	}
+
+	mr := merkleVerifier{
+		v: merkleRootsVerifier,
+	}
+
 	handler := &ArcDefaultHandler{
-		TransactionHandler:  transactionHandler,
-		MerkleRootsVerifier: merkleRootsVerifier,
-		NodePolicy:          policy,
-		logger:              logger,
-		now:                 time.Now,
-		peerRpcConfig:       peerRpcConfig,
-		apiConfig:           apiConfig,
+		TransactionHandler: transactionHandler,
+		NodePolicy:         policy,
+		logger:             logger,
+		now:                time.Now,
+		txFinder:           &finder,
+		mrVerifier:         mr,
```

Comment:
Why the pointer `&finder` here?

Reply:
Because it's a 41-byte struct, note that `ArcDefaultHandler` is used directly in its method, so I thought it would be beneficial to use a pointer here.